crypto/rsa/rsa_pmeth.c

```diff
@@ -432,9 +432,16 @@ static int pkey_rsa_ctrl(EVP_PKEY_CTX *ctx, int type, int p1, void *p2)
         } else {
             if (p1 < -2)
                 return -2;
-            if (rsa_pss_restricted(rctx) && p1 < rctx->min_saltlen) {
-                RSAerr(RSA_F_PKEY_RSA_CTRL, RSA_R_PSS_SALTLEN_TOO_SMALL);
-                return 0;
+            if (rsa_pss_restricted(rctx)) {
+                if (p1 == -2 && ctx->operation == EVP_PKEY_OP_VERIFY) {
```

Comment:
Yes we could do with some added defines for this and tidy up the ambiguity (-2 does different things on sign and verify).